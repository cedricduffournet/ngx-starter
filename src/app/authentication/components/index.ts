export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
