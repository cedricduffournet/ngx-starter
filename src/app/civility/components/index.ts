export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
