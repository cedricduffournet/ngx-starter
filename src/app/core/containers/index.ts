export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
