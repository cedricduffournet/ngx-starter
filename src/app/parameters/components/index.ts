export * from './referenced-table-menu/referenced-table-menu.component';
