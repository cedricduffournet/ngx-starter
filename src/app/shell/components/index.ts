export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
