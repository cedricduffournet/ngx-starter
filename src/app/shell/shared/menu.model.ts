export interface Menu {
  name: string;
  route: string[];
  roles: string[];
  cssClass?: string;
  faIcon?: string[];
}

export const appMenu: Menu[] = [
  {
    name: 'HOME',
    faIcon: ['fa', 'home'],
    cssClass: 'pt-3',
    route: ['dashboard'],
    roles: ['ROLE_USER']
  },
  {
    name: 'PARAMETERS',
    faIcon: ['fa', 'sliders-h'],
    cssClass: 'pt-3 mt-3 border-top border-dark',
    route: ['parameters'],
    roles: [
      'ROLE_COMPANY_VIEW_ALL',
      'ROLE_USER_VIEW',
      'ROLE_GROUP_VIEW',
      'ROLE_CIVILITY_VIEW'
    ]
  }
];
