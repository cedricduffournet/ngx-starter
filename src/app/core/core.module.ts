import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
