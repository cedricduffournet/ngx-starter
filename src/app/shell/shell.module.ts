import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { LogoModule } from '@app/shared/logo/logo.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { ActionsItemsModule } from '@app/shared/actions-item/actions-item.module';
import { IconsModule } from '@app/shared/icons/icons.module';

import {
  NavbarComponent,
  DropdownNavbarComponent,
  MenuComponent
} from '@app/shell/components';

@NgModule({
  imports: [
    CommonModule,
    RouterModule,
    TranslateModule,
    LogoModule,
    LangSwitcherModule,
    ActionsItemsModule,
    IconsModule
  ],
  exports: [NavbarComponent],
  declarations: [NavbarComponent, DropdownNavbarComponent, MenuComponent]
})
export class ShellModule {}
