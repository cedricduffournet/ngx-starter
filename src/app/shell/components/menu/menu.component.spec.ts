import { Component } from '@angular/core';
import { TestBed, ComponentFixture, async } from '@angular/core/testing';
import { Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';

import { IconsModule } from '@app/shared/icons';
import { TranslateModule } from '@ngx-translate/core';
import { MenuComponent } from '@app/shell/components';

@Component({
  template: ''
})
class StubComponent {}

describe('MenuComponent', () => {
  let fixture: ComponentFixture<MenuComponent>;
  let component: MenuComponent;
  let router: Router;
  const menu = [
    {
      name: 'HOME',
      faIcon: ['fa', 'home'],
      cssClass: 'pt-3',
      route: ['dashboard'],
      roles: ['ROLE_USER']
    },
    {
      name: 'PARAMETERS',
      faIcon: ['fa', 'sliders-h'],
      cssClass: 'pt-3 mt-3 border-top border-dark',
      route: ['parameters'],
      roles: [
        'ROLE_COMPANY_VIEW_ALL',
        'ROLE_USER_VIEW',
        'ROLE_GROUP_VIEW',
        'ROLE_CIVILITY_VIEW'
      ]
    }
  ];
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        TranslateModule.forRoot(),
        RouterTestingModule.withRoutes([
          {
            path: 'dashboard',
            component: StubComponent
          },
          {
            path: 'parameters',
            component: StubComponent
          }
        ]),
        IconsModule
      ],
      declarations: [MenuComponent, StubComponent]
    });
    fixture = TestBed.createComponent(MenuComponent);
    component = fixture.componentInstance;
    router = TestBed.get(Router);
  });

  it('should be created', () => {
    component.menu = menu;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit selectItem when click item in menu', async(() => {
    component.menu = menu;
    spyOn(component.selectItem, 'emit');

    if (fixture.ngZone) {
      // check link bellow for more info about this :
      // https://github.com/angular/angular/issues/25837
      fixture.ngZone.run(() => {
        fixture.detectChanges();
        fixture.whenStable().then(() => {
          const menus = fixture.nativeElement.querySelectorAll('a');
          const menuItem = menus[0];
          menuItem.click();
          expect(component.selectItem.emit).toHaveBeenCalledWith(menu[0]);
        });
      });
    }
  }));
});
