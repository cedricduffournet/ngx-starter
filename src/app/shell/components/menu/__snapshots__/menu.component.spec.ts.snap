// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent should be created 1`] = `
<app-menu
  menu={[Function Array]}
  selectItem={[Function EventEmitter]}
>
  <ul
    class="pl-0 menu"
  >
    
    <li>
      <a
        class="text-white d-block py-2 pt-3"
        href="/dashboard"
        ng-reflect-router-link="dashboard"
        ng-reflect-router-link-active="active"
        routerlinkactive="active"
      >
        <fa-layers
          class="fa-layers fa-fw"
          ng-reflect-fixed-width="true"
        >
          <fa-icon
            class="ng-fa-icon"
            ng-reflect-icon-prop="fa,home"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-home fa-w-18"
              data-icon="home"
              data-prefix="fa"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                fill="currentColor"
              />
            </svg>
          </fa-icon>
        </fa-layers>
         HOME 
      </a>
    </li>
    <li>
      <a
        class="text-white d-block py-2 pt-3 mt-3 border-top border-dark"
        href="/parameters"
        ng-reflect-router-link="parameters"
        ng-reflect-router-link-active="active"
        routerlinkactive="active"
      >
        <fa-layers
          class="fa-layers fa-fw"
          ng-reflect-fixed-width="true"
        >
          <fa-icon
            class="ng-fa-icon"
            ng-reflect-icon-prop="fa,sliders-h"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-sliders-h fa-w-16"
              data-icon="sliders-h"
              data-prefix="fa"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
                fill="currentColor"
              />
            </svg>
          </fa-icon>
        </fa-layers>
         PARAMETERS 
      </a>
    </li>
  </ul>
</app-menu>
`;
